server/src/main/java/org/cloudfoundry/identity/uaa/oauth/KeyInfo.java

```diff
@@ -161,11 +174,12 @@ class RsaKeyInfo extends KeyInfo {
     private Signer signer;
     private SignatureVerifier verifier;
     private String verifierKey;
+    private String verifierCertificate;
 
     public RsaKeyInfo(String keyId, String signingKey, String keyUrl) {
-        this(keyId, signingKey, keyUrl, null);
+        this(keyId, signingKey, keyUrl, null, null);
```

Comment:
This class is hard to understand with the potential null fields. 

Could we consider maybe extending this class, and adding the optional fields in the child class?

At the very least, I would prefer the nullable fields to be made [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) instead.

Reply:
added method